proposals/devcontainer-features.md

```diff
@@ -0,0 +1,143 @@
+
+## Folder Structure 
+
+A feature is a self contained entity in a folder. A feature release would be a tar file that contains all files part of the feature.
+
++-- feature
+|    +-- feature.json
+|    +-- validate.sh (default)
+|    +-- aquire.sh (default)
+|    +-- install.sh (default)
+|    +-- (other files)
+
+In case `feature.json`  does not include a reference for the lifecycle scripts the application will look for the default script names and will execute them if available.
+
+In case there is intent to create a set of features that share code, it is possible to create a feature collection in the following way:
+
+collectionFolder
++-- feature-collection.json
++-- common (or similar)
+|    +-- (library files)
++-- feature1
+|    +-- feature.json
+|    +-- install.sh (or others)
+|    +-- (other files)
++-- feature2
+(etc)
+
+## feature.json properties
+
+the `feature.json` file defines information about the feature to be used by any tool that helps a user utilize them, and the way the feature will be executed.
+
+The properties of the file are as follows:
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the feature/definition. The id should be unique in the context of the repository/published package where they exist. |
+| name | string | Name of the feature/definition |
+| description | string | Description of the feature/definition |
+| documentationURL | string | Url that targets the documentation of the feature. |
+| licenseURL | string | Url that points to the license of the feature. |
+| version | string | Version of the feature. |
+| metadata | any | Freeform data added by users for their own purposes. |
+| keywords | array | List of keywords relevant to a user that would search for this definition/feature. |
+| install.app | string | App to execute.ll (Optional, defaults to /bin/sh) |
+| install.file | string | Parameters/script to execute. (Defaults to validate.sh) |
+| options | object | Possible options to be passed as environment variables to the execution of the scripts |
+| containerEnv | object | A set of name value pairs that sets or overrides environment variables. |
+| privileged | boolean | If preveleged mode is required by the feature. |
+| init | boolean | If its necesarry to run `init`. |
+| capAdd | array | Additional capabilities needed by the feature. |
+| securityOpt | array | Security options needed by the feature. |
+| entrypoint | string | Set if the feature sets requires an entrypoint. |
+
+Options
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | Id of the option |
+| id.type | string | Type of the option |
+| id.enum | string array | possible values |
+| id.default | string | default value for the option |
+| id.description | string | Description for the option |
+
+## feature-collection.json properties.
+
+A feature collection file is a compilation of the `feature.json` files for each individual feature.
```

Comment:
Does it inline the same information or just reference each feature file?

Reply:
Inlines he information so that its the only file needed to show information to the user.